### PR TITLE
Fix time stamps

### PR DESCRIPTION
**Pull request by nithinraok**
* Update timestamps to calculate only once
* introduce segmentation module
* support multi-batch inference for speaker embedding extraction

### CODE REVIEW

**Comment by okuchaiev** [comment]
is this a bugfix? should this go to r1.2.0?

**Comment by nithinraok** [comment]
not a bug fix.

File: nemo/collections/asr/parts/utils/speaker_utils.py [hunk 1]
```diff
@@ -395,8 +394,25 @@ def write_rttm2manifest(paths2audio_files, paths2rttm_files, manifest_file):
...
 ):
+    """
+    Generate subsegments manifest from segments manifest file
+    Args
+    input:
+        segments_manifest file (str): path to segments manifest file, typically from VAD output
+        sub_segments_manifest_file (str): path to output sub segments manifest file (default (None) : writes to current working directory)
+        window (float): window length for segments to sub segments division
+        shift (float): hop length for sub segments shift 
+        min_subsegments_duration (float): exclude sub segments smaller than this duration value
```
Comment (tango4j) [inline nemo/collections/asr/parts/utils/speaker_utils.py:411]: This is very minor point, but it's better to be no space between "sub" and "segment"
https://www.dictionary.com/browse/subsegment

"subsegment"
This is for the sake of supporting search of the specific word.

**Review by tango4j — COMMENTED** [review]
All in all, there seems to be no major issue but it would be better off we use consistent variable name over the whole codebase.

sub_segment, sub segment ---> subsegment 

for both doc string and variables.

File: examples/speaker_recognition/conf/speaker_diarization.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 name: &name "ClusterDiarizer"
 
 num_workers: 4
-sample_rate: &sample_rate 16000
+sample_rate: 16000
```
Comment (titu1994) [inline examples/speaker_recognition/conf/speaker_diarization.yaml:4]: These two args should be inside model but upto you.

**Review by titu1994 — COMMENTED** [review]
Overall looks fine, perform the requested changes and then @tango4j for final review

**Review by tango4j — APPROVED** [review]
Seems like the required changes are all reflected.

**Pull request merged**